libnavigation-base/src/main/java/com/mapbox/navigation/base/internal/SDKRouteParser.kt

```diff
@@ -0,0 +1,58 @@
+package com.mapbox.navigation.base.internal
+
+import androidx.annotation.VisibleForTesting
+import com.mapbox.bindgen.DataRef
+import com.mapbox.bindgen.Expected
+import com.mapbox.navigation.base.internal.utils.mapToNativeRouteOrigin
+import com.mapbox.navigation.base.route.RouterOrigin
+import com.mapbox.navigator.RouteInterface
+import com.mapbox.navigator.RouteParser
+
+interface SDKRouteParser {
+    fun parseDirectionsResponse(
+        response: String,
+        request: String,
+        routerOrigin: RouterOrigin,
+    ): Expected<String, List<RouteInterface>>
+
+    fun parseDirectionsResponse(
+        response: DataRef,
+        request: String,
+        routerOrigin: RouterOrigin,
+    ): Expected<String, List<RouteInterface>>
+
+    companion object {
+        var default: SDKRouteParser = NativeRouteParserWrapper()
+            @VisibleForTesting set
```

Comment:
Why do you need to make it var with visible-for-testing setter? You can simply mock the object.
Then you can make it val, remove unused setter and the `resetDefault` method.

Reply:
I'm trying to avoid performance penalty static mocking by mockk brings.

Reply:
Static mocking is only done in tests while unused methods stay in our production code, increase aar size, etc.

Reply:
@dzinad , @DzmitryFomchyn , it seems that static mocking here slows UI unit tests for ~2 minutes. I see numbers only from 2 runs, so maybe it's just a misfortunate, but still.

Reply:
however I don't mind using mocking here, 20 or 22 minutes isn't a big difference 🙂